### PR TITLE
Starboard module

### PR DESCRIPTION
**Pull request by FjorgeTheHuman**
`en_kawaii` translations coming soon.

Ran through rubocop (with default settings), the only offense is:
```
modules/starboard.rb:9:1: C: Metrics/BlockLength: Block has too many lines. [46/25]
QBot.bot.reaction_add do |event| ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

There is not much I can do about it however, since it is where the core functionality of the module resides.

### CODE REVIEW

File: lib/db/schema.rb [hunk 1]
```diff
@@ -78,6 +78,12 @@ def self.define_schema
         t.timestamps
       end
 
+      create_table :starboard_entries do |t|
+        t.integer :server_id, null: false
+        t.integer :message_id, null: false, unique: true
+        t.integer :starboard_id, null: false, unique: true
```
Comment (anna328p) [inline lib/db/schema.rb:84]: add timestamps

File: lib/locales/en.yml [hunk 2]
```diff
@@ -285,6 +286,32 @@ en:
 
         success: 'Property `%s` of snippet `%s` is now `%s`.'
 
+    starboard:
+      help:
+        emoji: 'set the starboard reaction'
+        minimum-reacts: 'set the amount of reactions required for starboard'
```
Comment (anna328p) [inline lib/locales/en.yml:292]: "set the number of reactions required to star a message"

File: lib/locales/en.yml [hunk 3]
```diff
@@ -285,6 +286,32 @@ en:
 
         success: 'Property `%s` of snippet `%s` is now `%s`.'
 
+    starboard:
+      help:
+        emoji: 'set the starboard reaction'
```
Comment (anna328p) [inline lib/locales/en.yml:291]: "set the starboard reaction emoji"

File: modules/admin/config.rb [hunk 4]
```diff
@@ -262,6 +263,73 @@ module Admin
...
       end
 
+    when 'starboard', 'sb'
+      subcmd = args.shift
+      case subcmd
+      when 'help', 'h', nil
+        Config.help_msg event, 'cfg starboard', %i[emoji minimum-reacts channel delete-msgs]
+      when 'emoji', 'e'
+        opt = args.shift
+        if %w[show s].include?(opt)
```
Comment (anna328p) [inline modules/admin/config.rb:273]: make this all a `case` statement

File: modules/admin/config.rb [hunk 5]
```diff
@@ -279,3 +347,4 @@ module Admin
   # rubocop: enable Metrics/BlockLength
 end
 # rubocop: enable Metrics/ModuleLength, Metrics/BlockNesting
+# vim: ts=2:sw=2:set expandtab
```
Comment (anna328p) [inline modules/admin/config.rb:350]: please remove these

File: modules/starboard.rb [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+module Starboard
+  extend Discordrb::Commands::CommandContainer
+end
+
+# Reaction event handler
+QBot.bot.reaction_add do |event|
+  scfg = ServerConfig[event.server.id]
+  break if scfg.modules_conf['disabled'].include? 'starboard'
+
+  # Define functions for embed
```
Comment (anna328p) [inline modules/starboard.rb:13]: move these out of the event handler

File: modules/starboard.rb [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  ## Get field text
+  def embed_fields(event)
+    [{
+      name: 'Message',
+      value: format('[#%<name>s](%<link>s)', name: event.channel.name, link: event.message.link)
+    }]
+  end
+
+  # Get variables from db
+  emoji_name = (scfg.options['starboard-emoji'] || QBot.config.global.starboard.emoji)
```
Comment (anna328p) [inline modules/starboard.rb:40]: pull these lines out into another method and use destructuring assignment to get the values in

**Review by anna328p — CHANGES_REQUESTED** [review]
there's more i need to go through, this is the initial set of changes.

also note that the starboard will eventually be replaced by a managed trigger system. not anytime soon though.

**Comment by FjorgeTheHuman** [comment]
There are no more rubocop offenses.

File: modules/admin/config.rb (hunk 5 above)
Comment (FjorgeTheHuman) [reply modules/admin/config.rb]: I will do this after all the other requested changes

**Pull request closed without merging**